crates/proof-of-sql/src/sql/proof_plans/sort_merge_join_exec.rs

```diff
@@ -0,0 +1,600 @@
+        //TODO: Make it possible for `g` to have multiple columns
+        let g_col_eval = builder.try_consume_final_round_mle_evaluation()?;
+        let sorted_combined_indexes_eval = builder.try_consume_final_round_mle_evaluation()?;
+        dbg!("5 steps done");
+        // 6. Membership checks - rows of the sort merge join
+        // are from rows of the left and right tables with multiplicities
+        // `n_c` and `n_d`.
+        verify_membership_check(
+            builder,
+            alpha,
+            beta,
+            left_one_eval,
+            res_one_eval,
+            left_col_evals,
+            &res_col_evals[..num_columns_left],
+        )?;
+        verify_membership_check(
+            builder,
+            alpha,
+            beta,
+            right_one_eval,
+            res_one_eval,
+            right_col_evals,
+            &res_col_evals[num_columns_left..],
+        )?;
+        // 7. Membership checks - ordered set union
+        let left_on_col_positions = self
+            .left_join_columns
+            .iter()
+            .map(|col| {
+                self.left_selected_column_ident_aliases
+                    .iter()
+                    .position(|(a, _b)| *a == col.column_id())
+                    .unwrap()
+            })
+            .collect::<Vec<_>>();
+        let right_on_col_positions = self
+            .right_join_columns
+            .iter()
+            .map(|col| {
+                self.right_selected_column_ident_aliases
+                    .iter()
+                    .position(|(a, _b)| *a == col.column_id())
+                    .unwrap()
+            })
+            .collect::<Vec<_>>();
+        let left_on_col_evals = left_on_col_positions
+            .iter()
+            .map(|&pos| left_col_evals[pos])
+            .collect::<Vec<_>>();
+        let right_on_col_evals = right_on_col_positions
+            .iter()
+            .map(|&pos| right_col_evals[pos])
+            .collect::<Vec<_>>();
+        if left_col_evals.len() > 1 {
+            return Err(ProofError::VerificationError {
+                error: "Join on multiple columns not supported yet",
+            });
+        }
+        let res_left_on_col_evals = left_on_col_positions
+            .iter()
+            .map(|&pos| res_col_evals[pos])
+            .collect::<Vec<_>>();
+        let res_right_on_col_evals = right_on_col_positions
+            .iter()
+            .map(|&pos| res_col_evals[pos + num_columns_left])
+            .collect::<Vec<_>>();
+        verify_membership_check(
+            builder,
+            alpha,
+            beta,
+            g_one_eval,
+            left_one_eval,
+            &[g_col_eval],
+            &left_on_col_evals,
+        )?;
+        verify_membership_check(
+            builder,
+            alpha,
+            beta,
+            g_one_eval,
+            right_one_eval,
+            &[g_col_eval],
+            &right_on_col_evals,
+        )?;
```

Comment:
This is the "multiplicity" check as well, but `n_a`/`n_b` need to be exposed as the internal multiplicities.